homeassistant/components/recollect_waste/sensor.py

```diff
@@ -84,13 +86,21 @@ def icon(self):
     def update(self):
         """Update device state."""
         try:
-            pickup_event = self.client.get_next_pickup()
+            pickup_event_array = self.client.get_pickup_events(
+                date.today(), date.today() + timedelta(weeks=4)
+            )
+        except recollect_waste.RecollectWasteException as ex:
+            _LOGGER.error("Recollect Waste platform error. %s", ex)
+        else:
+            pickup_event = pickup_event_array[0]
+            next_pickup_event = pickup_event_array[1]
+            next_date = str(next_pickup_event.event_date)
             self._state = pickup_event.event_date
             self._attributes.update(
                 {
                     ATTR_PICKUP_TYPES: pickup_event.pickup_types,
                     ATTR_AREA_NAME: pickup_event.area_name,
+                    ATTR_NEXT_PICKUP_TYPES: next_pickup_event.pickup_types,
+                    ATTR_NEXT_PICKUP_DATE: next_date,
```

Comment:
If this is the only place `next_date` is used, why not inline its value?

Reply:
Formatting / readability, to prevent using str() function in the attribute update. Not tied to either, just seems cleaner the way it is, but can be argued either way.